include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1510,7 +1559,7 @@ __pattern_partial_sort_copy(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&&
 
         auto __buf_first = __buf.get();
 
-        auto __buf_last = __pattern_walk2</*_IsSync=*/::std::false_type>(
+        auto __buf_last = __pattern_walk2<__sync_mode>(
```

Comment:
I believe this can stay `__async_mode` based on the comment below.  
The temporary buffer will order the kernels appropriately, and no explicit wait is required. (Also, since `__buf_last` does not depend on the result of the kernel, it is a fixed increment of `__buf_first`)

Reply:
Fixed